cider-velox/src/ciderTransformer/CiderPlanUtil.cpp

```diff
@@ -42,7 +44,29 @@ std::shared_ptr<CiderPlanNode> CiderPlanUtil::toCiderPlanNode(
 std::shared_ptr<CiderPlanNode> CiderPlanUtil::toCiderPlanNode(
     VeloxNodeAddrPlanSection& planSection,
     VeloxPlanNodeAddrList& srcList) {
-  VELOX_UNSUPPORTED("multi-source PlanSection rewrite is not supported for now.");
+  std::shared_ptr<VeloxPlanFragmentToSubstraitPlan> v2SPlanFragmentConvertor_ =
+      std::make_shared<VeloxPlanFragmentToSubstraitPlan>();
+
+  std::shared_ptr<::substrait::Plan> sPlan =
+      std::make_shared<::substrait::Plan>(v2SPlanFragmentConvertor_->toSubstraitPlan(
+          planSection.target.nodePtr, planSection.source.nodePtr));
+
+  int32_t srcBranchId = planSection.source.branchId;
+  int32_t leftSrcBranchId = PlanBranches::getLeftSrcBranchId(srcBranchId);
+  int32_t rightSrcBranchId = PlanBranches::getRightSrcBranchId(srcBranchId);
+  auto newSrcMap = PlanUtil::toNodeAddrMap(srcList);
+  auto newLeftSrc = PlanUtil::findInNodeAddrMap(newSrcMap, leftSrcBranchId, 0);
+  auto newRightSrc = PlanUtil::findInNodeAddrMap(newSrcMap, rightSrcBranchId, 0);
+  if (newLeftSrc.first && newRightSrc.first) {
+    auto ciderPlanNode = std::make_shared<CiderPlanNode>(planSection.target.nodePtr->id(),
+                                                         newLeftSrc.second,
+                                                         newRightSrc.second
+                                                         planSection.target.nodePtr->outputType(),
+                                                         *sPlan);
+    return ciderPlanNode;
+  } else {
+    return nullptr;
```

Comment:
Any case we have condition ```newLeftSrc.first && newRightSrc.first``` stands false? If so, should we log an error msg here?